src/main/java/org/opensearch/knn/index/Parameter.java

```diff
@@ -204,6 +205,71 @@ public ValidationException validateWithData(Object value, VectorSpaceInfo vector
         }
     }
 
+    /**
+     * Double method parameter
+     */
+    public static class DoubleParameter extends Parameter<Double> {
+        public DoubleParameter(String name, Double defaultValue, Predicate<Double> validator) {
+            super(name, defaultValue, validator);
+        }
+
+        public DoubleParameter(
+            String name,
+            Double defaultValue,
+            Predicate<Double> validator,
+            BiFunction<Double, VectorSpaceInfo, Boolean> validatorWithData
+        ) {
+            super(name, defaultValue, validator, validatorWithData);
+        }
+
+        @Override
+        public ValidationException validate(Object value) {
+            ValidationException validationException = null;
+            if (value.equals(0)) value = 0.0;
+
+            if (!(value instanceof Double)) {
+                validationException = new ValidationException();
```

Comment:
I see this code is getting duplicated 

 ```
validationException = new ValidationException();
                validationException.addValidationError(
                    String.format(Locale.ROOT, "Value not of type Double for Double " + "parameter \"%s\".", getName())
                );
                return validationException;
```

Let's create a private function createValidationException which takes string and return exception.